### PR TITLE
[PUBDEV-4050] Remove support of Java6 from our code

### PR DESCRIPTION
**Pull request by mmalohlava**
This change only remove Java6 transformation from our build chain.
This will cause failures of Java6 test machines, but rest of jenkins tests
should not be affected.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -4,10 +4,8 @@ profile=Java
 # Branch codename (filled by releases)
 codename=bleeding_edge
 # Run findbugs
-runFindbugs=false
-# Produce java6 bytecode only if running on top of jenkins and JAVA_6_HOME is available
-doJava6Bytecode=auto
-# Run animal sniffer - by default false, but if java6 bytecode is requested 
+doFindbugs=false
+# Run animal sniffer - by default false, but if java6 bytecode is requested
```
Comment (st-pasha) [inline gradle.properties]: do we still need animal sniffer then?

**Comment by tomkraljevic** [comment]
you’d still want to sniff that java8 bytecodes don’t creep in.
so probably yes.

File: gradle.properties (hunk 1 above)
Comment (mmalohlava) [inline gradle.properties]: yes, i would like to keep animal sniffer inside - it could be configured to check against Java7 API compatibility

**Comment by mmalohlava** [comment]
Hi all, 

let's remove Java 6!  

Pls, if you are aware of any problems with removing Java6 pls block this PR else i will consider no reaction as a "GO" signal.

**Comment by ledell** [comment]
@mmalohlava This looks good.  I have found a fix that should fix the CRAN MacOS test machine issue.  It's written up in this JIRA and assigned to @navdeep-G https://0xdata.atlassian.net/browse/PUBDEV-4180   We can merge this in and complete this ticket later, or we can do this first, up to you.

**Comment by ledell** [comment]
@mmalohlava @navdeep-G Is this still going to get merged, or did the removal of Java6 happen in another PR (I thought we had just recently removed it). Also, all the tests are failing.

**Comment by mmalohlava** [comment]
@ledell yes, this is going to be merged. But needs one more fix.

**Comment by mmalohlava** [comment]
i just updated this PR with rebase on top of master.

**Pull request merged**